core/src/components/datetime/datetime.tsx

```diff
@@ -389,14 +404,15 @@ export class Datetime implements ComponentInterface {
   }
 
   /**
-   * Resets the internal state of the datetime
-   * but does not update the value. Passing a value
-   * ISO-8601 string will reset the state of
-   * te component to the provided date.
+   * Resets the internal state of the datetime to `internalState`
+   * and optionally updates the value if `updateValue` is true.
+   * Passing a valid ISO-8601 string will reset the state of
+   * the component to the provided date.
    */
   @Method()
-  async reset(value?: string) {
-    this.processValue(value);
+  async reset(internalState?: string, updateValue = false) {
```

Comment:
What if developers want to set the `value` to `undefined` but reset the internal state to a particular date?

Reply:
Hmm, `reset` could take two values instead?

`reset(internalState?: string, componentValue?: string)`